### ISSUE TITLE
Docs suggest invalid configuration combination

### ISSUE THREAD

**Issue by burgholzer** [pypa/cibuildwheel#1887]
### Description

The docs currently suggest that the new `build[uv]` feature can be used with extra arguments, such as
```toml
build-frontend = { name = "build[uv]", args = ["--no-isolation"] }
```

However, the above code will trigger an error
```
python -m build: error: argument --installer: not allowed with argument --no-isolation/-n
```

I just wanted to raise this here because the docs suggest this as a valid configuration. However, if you feel that it would make more sense to open an issue over at the https://github.com/pypa/build repo, given that it should actually be valid to request turning off build isolation when using build with the `uv` installer, please say so.

**Comment by henryiii** [comment]
Ohh, interesting, hadn’t thought about that. We probably should special case this.

**Comment by joerick** [comment]
Fixed in #1889.
